h2o-algos/src/main/java/hex/tree/FriedmanPopescusH.java

```diff
@@ -118,11 +117,11 @@ static double findFValue(long i, int[] currCombination, Frame currFValues, Frame
         double[] valueToFindFValueFor = getValueToFindFValueFor(currCombination, filteredFrame, i);
         String[] currNames = getCurrCombinationNames(currCombination, filteredFrame.names());
         FindFValue findFValueTask = new FindFValue(valueToFindFValueFor, currNames, currFValues._names, 1e-5);
-        Frame result = findFValueTask.doAll(Vec.T_NUM, currFValues).outputFrame();
-        if (result.numRows() == 0) {
+        Double result = findFValueTask.doAll(currFValues).result;
```

Comment:
We used to create a frame that could contain a lot of entries and then we took the first value from the first column. So I'm limiting the calculation here to be sure to have the first value but it can skip calculation of the rest. To make it easier I calculate the first value in each `map` and then using `reduce` I pick the value that is the first.